### ISSUE TITLE
infinite loop bug + solution

### ISSUE THREAD

**Issue by xunlight** [codemix/yii2-localeurls#40]
Firstly thank you very much for sharing your solution.

I've got infinite redirect loop.
You can fix infinite loop by insert something like this on line 403
```PHP
        //don't redirect when we are on same page
        if( $redirectUrl===$_SERVER['REQUEST_URI'] )
            return;
```
 (before redirect)
```PHP
Yii::$app->getResponse()->redirect($redirectUrl);
```

**Comment by mikehaertl** [comment]
Can you please give more details? How can I reproduce your issue? I.e. what is the configuration and which URL did you use?

**Comment by xunlight** [comment]
Firstly I've used for skeleton kartik's practical template>
https://github.com/kartik-v/yii2-app-practical

And this settings I have currently for urlManager
```PHP
        'urlManager' => [
            'class' => 'codemix\localeurls\UrlManager',
            'enablePrettyUrl'           => true,
            'showScriptName'            => false,
            'enableLanguageDetection'   => false,
            'enableLanguagePersistence' => false,
            'languages' => ['en'=>'en-US', 'dk'=>'da-DK', 'cz'=>'cs-CZ'],
            
        ]
```
and language settings
```PHP
    'language'       => 'da-DK',
    'sourceLanguage' => 'en-US',
```

**Comment by mikehaertl** [comment]
Which URL do you call that leads to the redirect? And can you maybe do some further debugging? The code should hopefully not be too hard to understand. The question is, why is `redirectToLanguage()` called at all? It should only call this method, if a redirect is really required.

**Comment by mikehaertl** [comment]
I tried with your configuration and can't reproduce your issue. So again: It would help if you do some debugging on your own.

**Comment by xunlight** [comment]
ok, I am going to do some debugging later today.

**Comment by xunlight** [comment]
Ok, I don't have now enough time, so this is what I know now.
It happen when it's enableLanguagePersistence = true;

And 
processLocaleUrl is for some reason called twice
and when I echo $pathInfo on line 231
It's for first time
"en/company/index"
and second time
"company/index"
and now it tries to redirect back to "en/company/index"

I have also in domain .dk which is same string as default language.

**Comment by mikehaertl** [comment]
Ok, thanks, that should help. I'll also try to have a look later. One question: Do you maybe have cookies disabled in your browser?

**Comment by xunlight** [comment]
No, I have enabled cookies, I was inspecting what I have in cookies.
I have enabled AdBlocker, but It shouldn't affect it.

**Comment by mikehaertl** [comment]
> processLocaleUrl is for some reason called twice

Is this during one request? Can't see, how this could happen. Are you sure, there's not maybe some AJAX requests involved, that send requests with different language code in the background or something?

It would really help, if you could provide a boiled down example setup, preferrably built on top of Yii's basic application template.

**Comment by xunlight** [comment]
yes, this is happening surelly during one request.
I've put "echo $pathInfo on line 231 " and "exit" before redirect.
And I've got echoed pathInfo two times...

I had day almost without computer today, I will do more tracing  tommorow.

**Comment by xunlight** [comment]
ok, I've got this.

I was reading route before initializing model >
```PHP
$route = Yii::$app->request->resolve();
```
and that function calls>
```PHP
Yii::$app->getUrlManager()->parseRequest($this); ... 
```
check https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L178

So on the end it's called twice...

and it's probable that some other user will use for example
```PHP
$route = Yii::$app->urlManager->parseRequest(Yii::$app->request);
```

**Comment by mikehaertl** [comment]
@xunlight Can you check if the above commit fixes your issue?

**Comment by xunlight** [comment]
yes, just checked. Your commit fixes this issue on my settings and source code.

**Comment by mikehaertl** [comment]
Great. I've released 1.2.4 which includes the fix. Thanks for your help!

**Comment by xunlight** [comment]
You are welcome. cheers
